rubygems-plugin/rubygems_plugin.rb

```diff
@@ -1,11 +1,23 @@
-# Yes borrowed from rbenv. Couldn't take my mind off that implementation
-
-Gem.post_install do |installer|
+module ReshimInstaller
+  def install(options)
+    super
+    # We don't know which gems were installed, so always reshim.
+    `asdf reshim ruby`
+  end
+end
 
-  if installer.spec.executables.any? && installer.bin_dir == Gem.default_bindir
+if defined?(Bundler::Installer)
```

Comment:
If this is defined, wouldn't we still want the `Gem` hooks setup too? It doesn't seem like these two things are mutually exclusive.

Reply:
I believe that `Bundler::Installer` is only defined when the `bundle` command is running. While Bundler is operating, we wouldn't want the gem hooks to run because we want to batch one reshim operation at the end rather than after each Gem.